numpy/core/src/umath/loops.c.src

```diff
@@ -1599,21 +1599,16 @@ FLOAT_@func@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSE
 NPY_NO_EXPORT NPY_GCC_OPT_3 void
 FLOAT_@func@_@isa@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(data))
 {
+    if (!run_unary_@isa@_@func@_FLOAT(args, dimensions, steps)) {
+        UNARY_LOOP {
 #if defined @CHK@ && defined NPY_HAVE_SSE2_INTRINSICS
-    @ISA@_@func@_FLOAT((npy_float*)args[1], (npy_float*)args[0], dimensions[0]);
+            @ISA@_@func@_FLOAT((npy_float *)op1, (npy_float *)ip1, 1);
```

Comment:
this is not correct, the existing logic is to dispatch to the vectorized code function if appropriate and if not run the fallback code inside the if statement.
a logic that may be unnecessarily convoluted, the simd code is in need of refactoring but that is a different issue. To fix your code just remove this line and the preprocessor directives around it.

the `NPY_GCC_OPT_3` macro can probably also be removed, the UNARY_LOOP macro does not allow for much optimization and to vectorize it with UNARY_LOOP_FAST gcc would require the `-ffast-math` which we intentionally do not use.